pkg/controller/topology.go

```diff
@@ -317,10 +359,17 @@ func aggregateTopologies(
 		})
 
 		// Pick the first node with topology keys
-		for _, csiNode := range csiNodes {
-			topologyKeys = getTopologyKeys(csiNode, driverName)
-			if topologyKeys != nil {
-				break
+		topologyKeys, err = getTopologyKeysFromCache(pvcNodeStore, pvcKeyForStore)
```

Comment:
It would be more efficient to check this first before we List + sort all the csinodes